src/transformState.js

```diff
@@ -5,7 +5,26 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+      case 'removeProperties':
+        for (const deleteItem of action.keysToRemove) {
+          delete state[deleteItem];
+        }
+        break;
+
+      case 'clear':
+        for (const clearItem in state) {
+          delete state[clearItem];
+        }
+        break;
+    }
```

Comment:
Pay attention
[CODE STYLE]: switch/case should always have default case for error handling.